### DISCUSSION TITLE
Andere Projekte

### DISCUSSION THREAD

**Discussion by pokeplayer2** [butterwecksolutions/Spotmarket-Switcher#82] [General]
Tolles Projekt und vielleicht steige ich als Tester ein, aber auf jeden Fall gibt es für Dich ein interessantes Projekt, welches viele Themen gelöst hat. https://www.photovoltaikforum.com/thread/142863-prognosebasierte-ladung-mittels-iobroker/?postID=1987334#post1987334 
- SOC wird beachtet
- die Sonne wird beachtet
- Grundlast wird beachtet
- Speicher Entladung wird beachtet.

Also alles, was man für die Kostenoptimierung benötigt ist dort drin, aber nicht in Shell. Ich suche nach so einer Lösung für Victron, ich weiß aber nicht, ob Du in diese Richtung aufrüsten willst. 

Wenn ja könnte, ich Sponsor (im Rahmen) und Tester sein ... programmieren nur minimal ;-)

**Comment by butterwecksolutions** [comment]
Hi,
eine Solarprognose ist schon im Spotmarket-Switcher eingebaut.
Außerdem werden noch die Speicherkosten sowie die Ladeverluste betrachtet bei der Entscheidung ob aufgeladen wird oder nicht.
Der SOC wird auch bereits ausgelesen, aber nur zur Information mitgeloggt.

Gruß!
Christian

**Reply by pokeplayer2** [reply]
Hört sich gut an, teste aber gerade die Victron - Lösung ... gestern spontan darüber gestolpert.

**Reply by butterwecksolutions** [reply]
Habe ich mir noch nicht angesehen, soll aber noch nicht ausgereift sein. Andere springen von der Victron Lösung wieder ab, siehe https://github.com/christian1980nrw/Spotmarket-Switcher/issues/81

**Comment by pokeplayer2** [comment]
Hi Christian, 
ich habe mir die Konfiguration angesehen und ein Punkt fehlt mir. Der 100%ige Support von dynamischen Preisen ... wir haben nur einen Stromanbieter Tibber, d.h. mit Wärmepumpe ist es immer nötig den Strom günstig einzukaufen und vorzuhalten. Dies unterstützt das Skript heute nicht so wie ich das sehe, weil man feste Grenzen vorgeben muss. Wenn man aber täglich ca. 40 kWh günstig beziehen muss, alleine schon für die WP ist dies mit dem Skript nur möglich, wenn der Strom gerade unter der Kostenschwelle liegt, die man fix einstellen muss. Ich will aber nicht jeden Tag diesen Wert einstellen, das soll das Skript machen. Vielleicht muss ich doch die IoBroker Lösung für Victron umwandeln, wenn das Victron zulässt. Die IoBroker Lösung nutzt heute Modbus "Befehle". Grüße und weiterhin frohes Schaffen.

**Reply by butterwecksolutions** [reply]
Hi,
ja momentan kann man nur die 6 günstigsten Stunden des Tages einzeln aktivieren oder eine Preisrange definieren.
Du hast Recht, das ist für manche Nutzer zu wenig.
Ich kann mir vorstellen das wir das entweder erweitern oder in einer Schleife umbauen wo man dann eine Stundenanzahl pro Tag eingibt und die günstigsten dann eingeschaltet werden.
Gruß!
Christian

**Comment by pokeplayer2** [comment]
Hi,
ich will die Lösung nicht komplett umbauen, aber in der anderen Lösung gibt man nur den Verbrauch mit Minimum SOC an und das Script rechnet aus, wann Strom bezogen werden muss. Das ist für den Nutzer, der seine Solaranlage schon länger hat und seinen Verbrauch kennt, am besten. Es wird einmal eingestellt und das Script macht den Rest ... kommt ein neuer Verbraucher dazu / es gibt eine Veränderung des Verbrauchs, stellt man den neuen Wert in kWh ein. Das heißt, über Monate fasst man die Lösung nicht an und vergisst, dass es sie gibt. 
Die Lösung wird auch nicht vom Einschalten von Lasten zu günstigen Zeiten großartig beeinflusst, weil das Entladen des Akkus zu Zeiten von günstigem Strom deaktiviert ist, diese Option kann man in den Sommermonaten deaktivieren, aber im Winter sinnvoll (Parameter mit Monaten von - bis wäre besser).
Gruß

**Comment by butterwecksolutions** [comment]
@pokeplayer2 Ich habe Dein Problem mit der längeren Laufzeit für z.B. Wärmepumpen mal hier adressiert, ich denke falls das geändert wird geht das in die Richtung.
https://github.com/christian1980nrw/Spotmarket-Switcher/issues/118

**Comment by butterwecksolutions** [comment]
@pokeplayer2 In der neuen Version im DEV sind mittlerweile alle genannten Features (außer die Betrachtung der Grundlast) enthalten. Ich schließe daher diesen Thread mal.
